api-definitions/platform/synthetic-data-generations.yaml

```diff
@@ -0,0 +1,104 @@
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/SDGJobBody'
+      responses:
+        "200":
+          description: A successfully submitted job
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+  /synthetic-data-generations/{job_id}:
+    get:
+      summary: Retrieve the status of a Synthetic Data Generation job
+      responses:
+        "200":
+          description: The status for the job
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+
+    delete:
+      summary: Cancel a running Synthetic Data Generation job
+      responses:
+        "200":
+          description: The status for the job after cancellation
+          content:
+            application/json:
+              schema:
+                $ref: '../common/job-status.yaml#/schemas/JobStatus'
+
+  /synthetic-data-generations/tasks:
+    get:
+      summary: List the currently supported SDG tasks
+      responses:
+        "200":
+          description: The set of currently supported SDG tasks
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/SDGTaskDefinition'
+
+components:
+  schemas:
+    SDGJobBody:
+      type: object
+      required: ['output_directory', 'tasks', 'seed_data']
+      properties:
+        output_directory:
+          description: Location to place the output in
+          $ref: '../common/file-pointer.yaml#/schemas/DirectoryPointer'
+
+        tasks:
+          description: Mapping from task name to task config. The config will be validated against the task's config schema.
+          type: object
+          additionalProperties:
+            type: object
+            description: Config for the given task. This will be deep-merged over the default values.
+
+        seed_data:
+          description: The file or directory containing the seed data
+          oneOf:
+            - $ref: '../common/file-pointer.yaml#/schemas/FilePointer'
+            - $ref: '../common/file-pointer.yaml#/schemas/DirectoryPointer'
+
+    SDGTaskDefinition:
+      type: object
+      required: ['name', 'data_json_schema', 'config_json_schema']
+      properties:
+        name:
+          type: string
+          description: The name of the task
+        data_json_schema:
+          type: object
+          description: The json schema for input data files for this task
+          # TODO: This doesn't render cleanly for some reason, but the body here
+          # must be a valid JSON Schema
+          # $ref: 'https://json-schema.org/draft-04/schema#'
+        data_example:
+          type: string
+          description: Example of an input data file for this task
+        config_json_schema:
```

Comment:
By "config" is this where you visualize a flexible "json blob" to where users could request "advanced parameters" when necessary to feed into SDG that maybe aren't default (lower level things like num samples, algorithm utilized in SDG, etc). 

I like the idea of it then starting out really flexible and I guess different implementations at potentially different moments would potentially only allow a given subset of what can be sent in the config level.

Reply:
Yep, that's exactly the idea here. We imagine the set of `tasks` to be extensible. Initially, this would be "build time" where the owner of the docker image would rebuild with new task implementations, but eventually we'd probably imagine users creating their own `tasks` by binding proprietary datasets/prompts/etc to existing generation algorithms. Each generation algorithm has a set of lower-level configs that can theoretically be overridden for each job, so the idea with this API is that when creating the job, the config overrides are an opaque blob, but you can query the system to understand the right schema for that blob beforehand. This avoids the need for us to keep a giant `oneOf` in the API definitions while still giving the user the ability to know the acceptable schemas that will be used for validation.